File: haushaltsbuch/src/main/java/org/github/haschi/haushaltsbuch/api/DeutschenWährungsbetragAnalysieren.java
package org.github.haschi.haushaltsbuch.api;

import javax.money.MonetaryAmount;
import javax.money.format.MonetaryAmountFormat;
import javax.money.format.MonetaryFormats;
import java.util.Locale;

final class DeutschenWährungsbetragAnalysieren
{

    private final MonetaryAmountFormat format;

    DeutschenWährungsbetragAnalysieren()
    {
        super();
        this.format = MonetaryFormats.getAmountFormat(Locale.GERMANY);
    }

    public MonetaryAmount aus(final String währungsbetrag)
    {
        return this.format.parse(währungsbetrag);
    }
}

File: haushaltsbuch/src/main/kotlin/org/github/haschi/haushaltsbuch/api/DeutschenWährungsbetragAnalysieren.kt
package org.github.haschi.haushaltsbuch.api

import javax.money.MonetaryAmount
import javax.money.format.MonetaryAmountFormat
import javax.money.format.MonetaryFormats
import java.util.Locale

internal class DeutschenWährungsbetragAnalysieren {

    private val format: MonetaryAmountFormat = MonetaryFormats.getAmountFormat(Locale.GERMANY)

    fun aus(währungsbetrag: String): MonetaryAmount {
        return this.format.parse(währungsbetrag)
    }
}
